### PR TITLE
docs: Add services configuration design doc

### LINKED ISSUE #16020: Add design doc for Services configuration

### PR DESCRIPTION
**Pull request by derektriley**
**Description**:

Adds a markdown file explaining the purpose of the various *Config classes.

**Related issue(s)**:

Fixes #16020 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/docs/design/services/service-configuration.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Service Configuration
+
+The bulleted list below provides a comprehensive overview of the various configuration settings used in the Hedera Services. These settings are defined in various Config classes,
+each managing a specific aspect of the system. For instance,
+`AccountsConfig.java` controls account-related properties, while
+`ContractsConfig.java` manages smart contract settings.
+This document serves as a valuable resource for understanding how
+Hedera Services can be customized and configured to meet specific requirements.
+For more detailed information on other types of configuration, please refer to the
+hedera-node/docs/design/services/service-configuration.md
```
Comment (thomas-swirlds-labs) [inline hedera-node/docs/design/services/service-configuration.md:10]: nit: small typo in target url leads to broken link (drop the /design)

File: hedera-node/docs/design/services/service-configuration.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+[Service Configuration](hedera-node/docs/services-configuration.md) file.
+
+- **AccountsConfig.java**: Provides configuration settings for various account-related properties, including admin roles and treasury accounts.
+- **ApiPermissionConfig.java**: Manages permissions for different API functionalities, specifying which accounts are allowed to execute certain operations.
+- **AutoCreationConfig.java**: Configures settings related to the automatic creation of accounts.
+- **AutoRenew2Config.java**: Contains settings for automatic renewal processes, specifying scan and renewal parameters.
+- **AutoRenewConfig.java**: Defines configurations for the auto-renewal feature, including target types for auto-renewal.
```
Comment (thomas-swirlds-labs) [inline hedera-node/docs/design/services/service-configuration.md:17]: I'm curious if these 2 AutoRenew Configs could be merged into one? 🤔

**Review by kimbor — DISMISSED** [review]

File: hedera-node/docs/design/services/service-configuration.md (hunk 2 above)
Comment (derektriley) [reply hedera-node/docs/design/services/service-configuration.md:16]: Going to just leave as two lines.

**Review by thomas-swirlds-labs — APPROVED** [review]
LGTM! thanks @derektriley

**Pull request merged**